File: Examples/CSharp/DrawingAndFormattingImages/DrawImagesUsingCoreFunctionality.cs
// GIST-ID: f5ca59c45061a4d61b08e3babf55e489
using Aspose.Imaging;
using Aspose.Imaging.ImageOptions;
using Aspose.Imaging.Sources;
using System;

/*
This project uses the Automatic Package Restore feature of NuGet to resolve the Aspose.Imaging for .NET API
when the project is built. Please see https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information.
If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET from https://releases.aspose.com/,
install it, and then add its reference to this project. For any issues, questions, or suggestions,
please feel free to contact us via https://forum.aspose.com/
*/

namespace Aspose.Imaging.Examples.CSharp.DrawingAndFormattingImages
{
    public class DrawImagesUsingCoreFunctionality
    {
        public static void Run()
        {
            Console.WriteLine("Running example DrawImagesUsingCoreFunctionality");
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_DrawingAndFormattingImages();

            // Create an instance of BmpOptions and set its various properties.
            BmpOptions imageOptions = new BmpOptions();
            imageOptions.BitsPerPixel = 24;

            // Create an instance of FileCreateSource and assign it to the Source property.
            imageOptions.Source = new FileCreateSource(dataDir + "DrawImagesUsingCoreFunctionality_out.bmp", false);

            // Create an instance of RasterImage and get the pixels of the image by specifying the
            // entire image bounds as the area.
            using (RasterImage rasterImage = (RasterImage)Image.Create(imageOptions, 500, 500))
            {
                Color[] pixels = rasterImage.LoadPixels(rasterImage.Bounds);
                for (int index = 0; index < pixels.Length; index++)
                {
                    // Set each pixel to yellow.
                    pixels[index] = Color.Yellow;
                }

                // Apply the pixel changes to the image and save all changes.
                rasterImage.SavePixels(rasterImage.Bounds, pixels);
                rasterImage.Save();
            }

            Console.WriteLine("Finished example DrawImagesUsingCoreFunctionality");
        }
    }
}

File: Examples/VisualBasic/DrawingAndFormattingImages/DrawImagesUsingCoreFunctionality.vb
Imports Aspose.Imaging.ImageOptions
Imports Aspose.Imaging.Sources

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https://Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'Install it and then add its reference to this project. For any issues, questions or suggestions 
'Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Aspose.Imaging.Examples.VisualBasic.DrawingAndFormattingImages
    Public Class DrawImagesUsingCoreFunctionality
        Public Shared Sub Run()
            ' ExStart:DrawImagesUsingCoreFunctionality
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_DrawingAndFormattingImages()

            ' Create an instance of BmpOptions and set its various properties
            Dim ImageOptions As New BmpOptions()
            ImageOptions.BitsPerPixel = 24

            ' Create an instance of FileCreateSource and assign it to Source property
            ImageOptions.Source = New FileCreateSource(dataDir & Convert.ToString("DrawImagesUsingCoreFunctionality_out.bmp"), False)

            ' Create an instance of RasterImage and Get the pixels of the image by specifying the area as image boundary
            Using rasterImage As RasterImage = DirectCast(Image.Create(ImageOptions, 500, 500), RasterImage)
                Dim pixels As Color() = rasterImage.LoadPixels(rasterImage.Bounds)
                For index As Integer = 0 To pixels.Length - 1
                    ' Set the indexed pixel color to yellow
                    pixels(index) = Color.Yellow
                Next

                ' Apply the pixel changes to the image and Save all changes.
                rasterImage.SavePixels(rasterImage.Bounds, pixels)
                rasterImage.Save()
            End Using
            ' ExStart:DrawImagesUsingCoreFunctionality
        End Sub
    End Class
End Namespace